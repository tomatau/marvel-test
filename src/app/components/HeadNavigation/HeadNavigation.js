import { NavLink } from 'react-router-dom'
import styles from './HeadNavigation.module.scss'

export default class HeadNavigation extends React.Component {
  render() {
    const { ...props } = this.props
    return (
      <nav className={styles.nav} {...props}>
        <NavLink exact activeClassName={styles.active} to='/'>
          Home
        </NavLink>
      </nav>
    )
  }
}
