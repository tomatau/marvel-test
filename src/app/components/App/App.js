import DocumentMeta from 'react-helmet'
import { Switch, Route } from 'react-router-dom'
import HeadNavigation from 'app/components/HeadNavigation/HeadNavigation'
import CodeSplit from 'app/components/CodeSplit'
// example s?css module import
import style from './App.module.scss'

const log = debug('App.js')

export default class App extends React.Component {
  render() {
    log('render')
    return (
      <div className={style.app}>
        <DocumentMeta
          defaultTitle='Marvel Test'
          titleTemplate='%s | Marvel Test'>
          <html lang='en' />
          <meta charSet='utf-8' />
          <meta name='viewport' content='width=device-width,initial-scale=1.0' />
          <meta name='description' content='Marvel Test, a minimal boilerplate for building universal react applications' />
          <meta name='keywords' content='react,redux,react-router,koa,universal,babel,es7,hmr,webpack' />
        </DocumentMeta>
        <HeadNavigation />
        <h1>Marvel Test</h1>
        <main className={style.content}>
          <Switch>
            <CodeSplitRoute
              exact
              path='/'
              load={() => import('app/routes/HomeRoute/HomeRoute')}
            />
            <CodeSplitRoute
              load={() => import('app/routes/NotFoundRoute/NotFoundRoute')}
            />
          </Switch>
        </main>
      </div>
    )
  }
}

const CodeSplitRoute = ({ load, ...props }) => (
  <Route {...props}
    render={() => (
      <CodeSplit load={load}>
        {(Comp) => Comp && <Comp />}
      </CodeSplit>
    )}
  />
)
