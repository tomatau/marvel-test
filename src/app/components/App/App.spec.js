import DocumentMeta from 'react-helmet'
import HeadNavigation from 'app/components/HeadNavigation/HeadNavigation'
import App from './App'
import { shallow } from 'enzyme'
import styles from './App.module.scss'

describe('App Component', function () {
  helpers.setupSnapshots(__filename)

  beforeEach(() => {
    this.tree = shallow(<App />)
  })

  it('renders a div tag with className at rootNode', () => {
    expect(this.tree.hasClass(styles.app)).to.eql(true)
  })

  it('renders a Helmet document meta as firtChild', () => {
    const firstChild = this.tree.childAt(0)
    expect(firstChild).to.have.type(DocumentMeta)
  })

  it('renders HeadNavigation as second child', () => {
    const secondChild = this.tree.childAt(1)
    expect(secondChild).to.have.type(HeadNavigation)
  })

  it('renders a h1 as third child', () => {
    const thirdChild = this.tree.childAt(2)
    expect(thirdChild).to.have.type('h1')
  })

  it('renders a main.content as fourth child', () => {
    const fourthChild = this.tree.childAt(3)
    expect(fourthChild).to.have.type('main')
    expect(fourthChild).to.have.className(styles.content)
  })

  describe('DocumentMeta', () => {
    it('sets the meta options', () => {
      expect(this.tree.find(DocumentMeta).node).to.eql(
        <DocumentMeta
          defaultTitle='Marvel Test'
          titleTemplate='%s | Marvel Test'>
          <html lang='en' />
          <meta charSet='utf-8' />
          <meta name='viewport' content='width=device-width,initial-scale=1.0' />
          <meta name='description' content='Marvel Test, a minimal boilerplate for building universal react applications' />
          <meta name='keywords' content='react,redux,react-router,koa,universal,babel,es7,hmr,webpack' />
        </DocumentMeta>
      )
    })
  })

  describe('Title', () => {
    it('should have "Marvel Test" as text child', () => {
      const title = this.tree.find('h1')
      expect(title.text()).to.contain('Marvel Test')
    })
  })

  describe('Main', () => {
    it('renders routes inside Main', () => {
      expect(
        snap(this.tree.find('main'))
      ).to.matchSnapshot()
    })
  })
})
