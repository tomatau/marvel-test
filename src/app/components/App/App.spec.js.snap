// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Main renders routes inside Main 1`] = `
<main
  className="src-app-components-App-App-module-content"
>
  <Switch>
    <CodeSplitRoute
      exact={true}
      load={[Function]}
      path="/"
    />
    <CodeSplitRoute
      load={[Function]}
    />
  </Switch>
</main>
`;
